front/lib/iam/users.ts

```diff
@@ -95,46 +89,46 @@ export async function maybeUpdateFromExternalUser(
 
 export async function createOrUpdateUser(
   session: SessionWithUser
-): Promise<{ user: UserType; created: boolean }> {
+): Promise<{ user: UserResource; created: boolean }> {
   const { user: externalUser } = session;
 
   const user = await fetchUserFromSession(session);
 
   if (user) {
+    const newUser = {} as Partial<UserType>;
     // We only update the user's email if the email is verified.
     if (externalUser.email_verified) {
-      user.email = externalUser.email;
+      newUser.email = externalUser.email;
     }
 
     // Update the user object from the updated session information.
-    user.username = externalUser.nickname;
-    user.name = externalUser.name;
+    newUser.username = externalUser.nickname;
 
     if (!user.firstName && !user.lastName) {
       if (externalUser.given_name && externalUser.family_name) {
-        user.firstName = externalUser.given_name;
-        user.lastName = externalUser.family_name;
+        newUser.firstName = externalUser.given_name;
+        newUser.lastName = externalUser.family_name;
       } else {
         const { firstName, lastName } = guessFirstAndLastNameFromFullName(
           externalUser.name
         );
-        user.firstName = firstName;
-        user.lastName = lastName;
+        newUser.firstName = firstName;
+        newUser.lastName = lastName;
       }
     }
 
-    await user.save();
+    await user.update({ ...newUser, createdAt: user.createdAt });
```

Comment:
Let's make it a specific method user.upateInfo maybe? So that we don't need that newUser weirdness

Reply:
💯 ☝️

Reply:
Not sure what you mean by that - resource attributes are readonly so I can't just say `user.email = x`
You'd prefer doing a bunch of updates than building a newUser object? Or maybe I'm not seeing how you want to proceed

Reply:
More like cut a UserResource.updateInfo({ name, ..., createdAt: ...})

And then create that argument object and use that method

Reply:
Done